src/commonMain/kotlin/com/github/quillraven/fleks/system.kt

```diff
@@ -179,4 +179,28 @@ abstract class IteratingSystem(
     companion object {
         private val EMPTY_COMPARATOR = EntityComparator { _, _ -> 0 }
     }
+
+    /**
+     * Any [IteratingSystem] having this interface will be triggered
+     * by own [Family] similarly to [Family.addHook].
+     */
+    interface FamilyOnAdd {
```

Comment:
Extract the interfaces from IteratingSystem because otherwise, when you autoimport them in IntelliJ, it gets imported as `IteratingSystem.FamilyOnAdd`. I prefer just `FamilyOnAdd`.

I guess your reasoning was that those interfaces should only be used in combination with an IteratingSystem but we anyway catch a wrong usage in the world configuration. So imo moving it outside and have the shorter import version automatically is better.